PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/springcloud/SpringCloudAppPropertyView.java

```diff
@@ -343,12 +345,38 @@ private void syncSaveStatus() {
             return;
         }
         try {
-            saveButton.setEnabled(MapUtils.isNotEmpty(getModifiedDataMap()));
+            Map<String, Object> map = getModifiedDataMap();
+            saveButton.setEnabled(MapUtils.isNotEmpty(map));
+            updateBorder(this.cpuCombo, map.containsKey(CPU));
+            updateBorder(this.memCombo, map.containsKey(MEMORY_IN_GB_KEY));
+            updateBorder(this.jvmOpsTextField, map.containsKey(JVM_OPTIONS_KEY));
+            updateBorder(this.javaVersionCombo, map.containsKey(JAVA_VERSION_KEY));
+            // updateBorder(this.envTable, map.containsKey(ENV_TABLE_KEY));
         } catch (IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
             PluginUtil.showErrorNotificationProject(project, "Cannot get property through reflection", e.getMessage());
         }
     }
 
+    private void updateBorder(final JComponent component, boolean highlight) {
+        if (highlight) {
+            addHighLight(component);
+        } else {
+            removeHighLight(component);
+        }
+    }
+
+    private void removeHighLight(final JComponent component) {
+        Border border = borderMap.remove(component);
+        if (border != null) {
+            component.setBorder(border);
+        }
+    }
+
+    private void addHighLight(final JComponent component) {
+        borderMap.putIfAbsent(component, component.getBorder());
```

Comment:
Why we use a map here. seems a List is enough.
Besides, ConcurrentHashMap will throw exception when null values, this may occurs for Jcomponent

Reply:
Fixed